### PR TITLE
Order Queue System

### PR DESCRIPTION
**Pull request by Vinyzu**
- Implemented Order Spawning, Checking
- Implemented Order Checking
- Implemented various User Input Validation Checks

Co-Authored-By: koviubi56 <koviubi56@duck.com>

### CODE REVIEW

**Comment by Vinyzu** [comment]
That better-late-than-never co-authoring only worked in our hearts...

File: src/sincere_singularities/modules/order_queue.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+from disnake.ext.commands.errors import CommandInvokeError
+
+from sincere_singularities.modules.order import Order
+from sincere_singularities.utils import generate_random_avatar_url
+
+
+class OrderQueue:
+    """The Class for managing the order queue. Orders can be spawned and deleted from here."""
+
+    orders: ClassVar[dict[str, tuple[Order, WebhookMessage]]] = {}
```
Comment (koviubi56) [inline src/sincere_singularities/modules/order_queue.py]: Why is this annotated as a ClassVar when it isn't? And where is this initiated?

File: src/sincere_singularities/modules/order_queue.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    async def discard_order(self, order_id: str) -> None:
+        """
+        Discard a specific Order by ID after its completed.
+
+        Args:
+            order_id: ID of the Order to discard.
+        """
+        if self.orders.get(order_id):
+            del self.orders[order_id]
```
Comment (koviubi56) [inline src/sincere_singularities/modules/order_queue.py]: [Ask forgiveness, not permission](https://stackoverflow.com/a/12270384/14892922)

File: src/sincere_singularities/modules/order_queue.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    webhook: Webhook
+    orders_thread: Thread
+
+    def __init__(self, inter: ApplicationCommandInteraction) -> None:
+        self.user = inter.user
+        self.channel = inter.channel
+
+    async def start_orders(self) -> None:
+        """Start the orders queue. Spawn a new Webhook and Order Thread"""
+        # Creating the Order Webhook
```
Comment (koviubi56) [inline src/sincere_singularities/modules/order_queue.py:23]: Maybe add a TODO?

File: src/sincere_singularities/modules/order_queue.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+        Args:
+            order_id: ID of the Order to discard.
+        """
+        if self.orders.get(order_id):
+            del self.orders[order_id]
+
+    async def stop_orders(self) -> None:
+        """Stop All Orders (when stopping the game)."""
+        with suppress(Exception):
```
Comment (koviubi56) [inline src/sincere_singularities/modules/order_queue.py:85]: Add a TODO here

**Review by koviubi56 — CHANGES_REQUESTED** [review]
Once you can fix these, it's good to go.

File: src/sincere_singularities/modules/order_queue.py (hunk 1 above)
Comment (Vinyzu) [reply src/sincere_singularities/modules/order_queue.py]: Ruff was mad at me for now using a ClassVar there.
I just added it to make it pass 🫠...

File: src/sincere_singularities/modules/order_queue.py (hunk 3 above)
Comment (Vinyzu) [reply src/sincere_singularities/modules/order_queue.py:23]: For what specifically?

Comment (koviubi56) [reply src/sincere_singularities/modules/order_queue.py:23]: nvm

File: src/sincere_singularities/modules/order_queue.py (hunk 1 above)
Comment (koviubi56) [reply src/sincere_singularities/modules/order_queue.py]: Where is it initiated?

**Review by koviubi56 — APPROVED** [review]
LGTM

**Pull request merged**